.github/workflows/release.yaml

```diff
@@ -116,3 +110,38 @@ jobs:
           # Remove 'v' prefix from version
           CHART_VERSION=${VERSION#v}
           helm push helm-releases/kepler-helm-${CHART_VERSION}.tgz oci://${{ env.IMG_BASE }}/charts
+
+  release:
+    needs: build
+    permissions:
+      # contents: write is required for publishing Github Releases
+      # This follows Github's recommended pattern: top-level read-all with minimal job-level permissions
+      # See: https://github.com/softprops/action-gh-release?tab=readme-ov-file#permissions
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Release Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: release-artifacts
+
+      - name: Extract version
+        shell: bash
+        id: version
+        run: |
+          TAG_NAME=${{ github.ref_name }}
+          echo "version=$TAG_NAME" >> "$GITHUB_OUTPUT"
+
+      - name: Create GitHub Release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{ github.ref_name }}
+          name: release-${{ steps.version.outputs.version }}
```

Comment:
release-${{ github.ref_name }} ?
if in this new design we just have bash here for ... value injection?